employee-management-client/src/app/components/typeahead/client-typeahead/client-typeahead.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { ClientService } from 'src/app/client-details/client.service';
+
+@Component({
+  selector: 'app-client-typeahead',
+  templateUrl: './client-typeahead.component.html',
+  styleUrls: ['./client-typeahead.component.scss']
+})
+export class ClientTypeaheadComponent implements OnInit {
+
+  selected?: ClientTypeahead;
+  dropDownList?: ClientTypeahead[];
+  dropDown?: boolean;
+  searchString?: string;
+
+  constructor(private clientServices: ClientService) { /* TODO document why this constructor is empty */  }
+
+  ngOnInit(): void {
+    this.selected = undefined;
+    this.dropDownList = [];
+    this.dropDown = false;
+    this.searchString = '';
+  }
+
+  onTypeaheadChanged(event: string) {
+    this.clientServices.searchClientData(event).subscribe((x: ClientTypeahead[])=>{
+      this.dropDownList = x;
+      console.log(this.dropDownList);
+      if(this.searchString?.length === 0) {
+        this.dropDownList = [];
+      }
+      if(this.dropDownList.length !== 0) {
```

Comment:
at least 3 characters need to be typed before hitting the API. Also give debouncing of 2000 milli seconds for typeahead.